### PR TITLE
Update the macOS installer and docs

### LINKED ISSUE #60858: [FEATURE REQUEST] change messaging in osx installer
opened this issue based on https://github.com/saltstack/salt/pull/60857#issuecomment-916243476

Thanks for opening the issue, I'm testing salt from new user prospective. In the process of installing salt on [macOS](https://repo.saltproject.io/#osx) The installer does it's job but the information mentioned on first screen should be on last screen also on the documentation pages because at the end of installation it doesn't tell where salt was installed. So, this should be corrected in installer & in documentation. 

image

Also, at the last screen of installer we ask user to try a command, it doesn't work as mentioned because looks like path is not set. So either we set the path through installer or change the documentation to guide users about navigating to install location, setting path and trying command.

image

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Updates the welcome and conclusion pages of the macOS installer
Brings some of the information in the welcome to the conclusion page
Adds clarification about the salt-config cli utility
Adds docs to the installation page for macOS for the installer
Adds docs to the installation page for macOS for the salt-config tool

### What issues does this PR fix or reference?
Fixes: #60858

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: doc/topics/installation/osx.rst [hunk 1]
```diff
@@ -29,26 +88,38 @@ Installation from Homebrew
...
+``/opt/local`` by default, with config files under ``/opt/local/etc/salt``. For
+best results, add ``/opt/local/bin`` to your PATH.
 
 .. code-block:: bash
 
     sudo port install salt
 
-Variants allow selection of python version used to run salt, defaulting to python27, but also supporting python34, python35, and python36. To install salt with Python 3.6, use the python36 variant, for example:
+Variants allow selection of the python version used to run Salt, defaulting to
+python27, but also supporting python34, python35, and python36. To install Salt
```
Comment (Ch3LL) [inline doc/topics/installation/osx.rst:111]: This is outdated information. We no longer support python27 or python34. Can you update what the default version is and remove python34?

File: doc/topics/installation/osx.rst [hunk 2]
```diff
@@ -65,16 +137,22 @@ When only using the macOS system's pip, install this way:
...
+
 .. note::
-    Salt master on macOS is not tested or supported by SaltStack. See `SaltStack Platform Support <https://saltstack.com/product-support-lifecycle/>`_ for more information.
 
-To run salt-master on macOS, sudo add this configuration option to the /etc/salt/master file:
+    Salt master on macOS is not tested or supported by SaltStack. See
+    `SaltStack Platform Support <https://saltstack.com/product-support-lifecycle/>`_
+    for more information.
+
+To run salt-master on macOS, sudo add this configuration option to the
```
Comment (Ch3LL) [inline doc/topics/installation/osx.rst:147]:
```suggestion
To run salt-master on macOS, add this configuration option to the
```

File: doc/topics/installation/osx.rst [hunk 3]
```diff
@@ -65,16 +137,22 @@ When only using the macOS system's pip, install this way:
...
+``/etc/salt/master`` file:
 
 .. code-block:: bash
 
     max_open_files: 8192
 
-On versions previous to macOS 10.10 (Yosemite), increase the root user maxfiles limit:
+On versions previous to macOS 10.10 (Yosemite), increase the root user maxfiles
```
Comment (Ch3LL) [inline doc/topics/installation/osx.rst:154]: We no longer support macOS10.10.  Can you remove this section?

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: doc/topics/installation/osx.rst (hunk 3 above)
Comment (twangboy) [reply doc/topics/installation/osx.rst]: Done

File: doc/topics/installation/osx.rst (hunk 2 above)
Comment (twangboy) [reply doc/topics/installation/osx.rst]: Done

File: doc/topics/installation/osx.rst (hunk 1 above)
Comment (twangboy) [reply doc/topics/installation/osx.rst]: Done, though the ports package states that it supports python38 on macOS.
https://ports.macports.org/search/?q=salt

**Review by Ch3LL — DISMISSED** [review]

**Comment by MKLeb** [comment]
Removing myself as a required reviewer, as I don't have enough knowledge to approve and @Ch3LL has already approved.  Looks like the test failure is due to old multimaster tests and is unrelated.

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**